kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -51,7 +53,7 @@ import org.apache.kyuubi.operation.log.OperationLog
 private[kyuubi] class EngineRef(
     conf: KyuubiConf,
     user: String,
-    primaryGroup: String,
+    groupProvider: GroupProvider,
```

Comment:
It seems also can be lazily initialized, not needed in the constructor?

Reply:
Thanks for your reply.

It seems that the groupProvider needs to obtain from KyuubiSessionManager. Whether it is passing KyuubiSessionManager through the constructor or trying to initialize and create one, the chaotic changes introduced and the cost paid may be greater than the current transformation.

Reply:
the cost is mostly test code refactoring, but I'm OK to keep the PR AS-IS.

Reply:
Ok, thank you